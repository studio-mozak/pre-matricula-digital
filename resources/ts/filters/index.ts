import { Address, GoogleMapsMarker } from '@/types';
import {
  getFormattedDate,
  getFormattedDateTime,
  getFormattedTime,
} from '@/datetime';
import {
  joinAddress,
  joinAddressLine1,
  joinAddressLine2,
  joinAddressLine3,
  markerAddress,
} from '@/util';

export interface Filters {
  formatDateTime: (date: string) => string;
  formatDate: (date: string) => string;
  formatTime: (date: string) => string;
  joinAddress: (address: Address) => string;
  joinAddressLine1: (address: Address) => string;
  joinAddressLine2: (address: Address) => string;
  joinAddressLine3: (address: Address) => string;
  markerAddress: () => GoogleMapsMarker;
}

const filters: Filters = {
  formatDateTime(date: string) {
    return getFormattedDateTime(date);
  },
  formatDate(date: string) {
    return getFormattedDate(date);
  },
  formatTime(date: string) {
    return getFormattedTime(date);
  },
  joinAddress,
  joinAddressLine1,
  joinAddressLine2,
  joinAddressLine3,
  markerAddress,
};

export default filters;
