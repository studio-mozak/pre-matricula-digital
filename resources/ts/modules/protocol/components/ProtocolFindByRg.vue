<template>
  <protocol-find v-model:student="student" type="RG">
    <template #default="slot">
      <x-field
        v-model="student.student_rg"
        container-class="col-12 col-md-6"
        name="student_rg"
        rules="required"
        label="RG do(a) aluno(a)"
        type="RG"
        :errors="!!slot.errors['student_rg']"
      />
    </template>
  </protocol-find>
</template>

<script setup lang="ts">
import ProtocolFind from '@/modules/protocol/components/ProtocolFind.vue';
import { Student } from '@/types';
import XField from '@/components/x-form/XField.vue';
import { ref } from 'vue';

const student = ref<Student>({
  student_rg: null,
  preregistrations: [],
});
</script>
