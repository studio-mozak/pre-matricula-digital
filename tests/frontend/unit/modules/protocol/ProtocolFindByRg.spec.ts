import { VueWrapper, mount } from '@vue/test-utils';
import ProtocolFindByRg from '@/modules/protocol/components/ProtocolFindByRg.vue';
import VTooltip from 'v-tooltip';
import filters from '@/filters';
import { expect, vi } from 'vitest';

vi.mock('vue-router/dist/vue-router.mjs', () => ({
  useRouter: () => ({
    push: spyPush,
  }),
}));

const spyPush = vi.fn();

const onMounted = async () => {
  const wrapper = mount(ProtocolFindByRg, {
    global: {
      stubs: ['x-field', 'router-link'],
      directives: {
        tooltip: VTooltip,
      },
      mocks: {
        $router: () => ({
          push: spyPush,
        }),
      },
      provide: {
        $filters: filters,
      },
    },
  });

  await wrapper.vm.$nextTick();

  return wrapper;
};

describe('ProtocolFindByRg', () => {
  let wrapper: VueWrapper<InstanceType<typeof ProtocolFindByRg>>;

  beforeEach(async () => {
    wrapper = await onMounted();

    await wrapper.vm.$nextTick();
  });

  afterEach(() => {
    vi.clearAllMocks();
  });

  test('the component exists', () => {
    expect(wrapper).toBeTruthy();
  });

  test('should match snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test('should fill "RG DO(A) ALUNO(A)" input successfully', async () => {
    const input = wrapper.findComponent({
      name: 'x-field-stub',
    });

    expect(input).toBeTruthy();

    wrapper.vm.student.student_rg = 'XXX123456XXX';

    expect(wrapper.vm.student.student_rg).toBe('XXX123456XXX');
  });
});
