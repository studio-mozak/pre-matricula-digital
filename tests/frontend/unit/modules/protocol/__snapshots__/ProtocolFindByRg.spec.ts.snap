// Vitest Snapshot v1

exports[`ProtocolFindByRg > should match snapshot 1`] = `
<div>
  <div>
    <form
      autocomplete="off"
      class="row"
    >
      
      
      
      <div
        class="col-12"
      >
        <div
          class="row"
        >
          <div
            class="col-md-8 mt-5 offset-md-2"
          >
            <h2
              class="font-muli-20-primary mb-4"
            >
               Identifique o(a) aluno(a) para encontrar o(s) protocolo(s): 
            </h2>
            <p
              class="text-justify font-hind"
            >
               Preencha os dados abaixo para identificar se o(a) 
              <strong>
                aluno(a)
              </strong>
               já possui alguma inscrição de pré-matrícula. 
            </p>
            <div
              class="d-flex justify-content-center mt-5 row"
            >
              
              <x-field-stub
                allowclear="false"
                bind="[object Object]"
                containerclass="col-12 col-md-6"
                disabled="false"
                errors="false"
                label="RG do(a) aluno(a)"
                loading="false"
                name="student_rg"
                options=""
                rules="required"
                searchable="false"
                type="RG"
              />
              
            </div>
            <!--v-if-->
          </div>
        </div>
      </div>
      
      <div
        class="col-12"
      >
        <div
          class="row mt-5"
        >
          
          <div
            class="offset-md-3 col-md-3"
          >
            <button
              class="x-btn x-btn-item non-selectable no-outline x-btn--outline x-btn--rectangle text-gray-500 border-gray-500 x-btn--actionable pmd-focusable pmd-hoverable x-btn--no-uppercase w-100"
              role="button"
              style="font-size: 20px;"
              tabindex="0"
              type="button"
            >
              <span
                class="x-btn__content text-center d-flex justify-content-center no-wrap text-no-wrap"
              >
                <span
                  class="block"
                >
                  Voltar
                </span>
              </span>
              <transition-stub
                appear="false"
                css="true"
                name="x-transition--fade"
                persisted="false"
              >
                <!---->
              </transition-stub>
            </button>
          </div>
          <div
            class="col-md-3 mt-3 mt-md-0"
          >
            <button
              class="x-btn x-btn-item non-selectable no-outline x-btn--standard x-btn--rectangle bg-primary text-white x-btn--actionable pmd-focusable pmd-hoverable x-btn--no-uppercase btn btn-block btn-primary w-100 h-100 flex-row justify-content-center"
              disabled="false"
              role="button"
              style="font-size: 20px;"
              tabindex="0"
              type="button"
            >
              <span
                class="x-btn__content text-center d-flex justify-content-center no-wrap text-no-wrap"
              >
                <span
                  class="block"
                >
                  Prosseguir
                </span>
              </span>
              <transition-stub
                appear="false"
                css="true"
                name="x-transition--fade"
                persisted="false"
              >
                <!---->
              </transition-stub>
            </button>
          </div>
          
        </div>
      </div>
    </form>
  </div>
  <!--v-if-->
</div>
`;
